### PR TITLE
Various cleanups and refactorings

### PR DESCRIPTION
**Pull request by tru**
While working on the 1.1 branch I have done some various cleanups
and refactorings.

### CODE REVIEW

**Comment by plexappbuilder** [comment]
:-1: Tests failed!

**Comment by plexappbuilder** [comment]
:-1: Tests failed!

File: src/input/InputComponent.cpp [hunk 1]
```diff
@@ -166,9 +176,6 @@ void InputComponent::remapInput(const QString &source, const QString &keycode, b
     return;
   }
 
-  // hide mouse if it's visible.
-  SystemComponent::Get().setCursorVisibility(false);
-
```
Comment (wm4) [inline src/input/InputComponent.cpp]: Why does this disappear?

Comment (wm4) [inline src/input/InputComponent.cpp]: Oh I see, it got moved in the previous commit. These should probably be squashed.

Comment (tru) [inline src/input/InputComponent.cpp]: ugh.

**Comment by plexappbuilder** [comment]
:balloon: - Good work!

**Comment by plexappbuilder** [comment]
:balloon: - Good work!

**Pull request merged**